src/autogpt_plugins/gradio-tools/__init__.py

```diff
@@ -0,0 +1,239 @@
+"""Twitter API integrations using Tweepy."""
+from typing import Any, Dict, List, Optional, Tuple, TypedDict, TypeVar
+from dotenv import load_dotenv
+from auto_gpt_plugin_template import AutoGPTPluginTemplate
+from pathlib import Path
+import os
+from .tools import (AutoGPTClipInterrogatorTool,
+                    AutoGPTStableDiffusion,
+                    AutoGPTWhisperTool,
+                    AutoGPTTextToVideoTool,
+                    AutoGPTCaptioner,
+                    AutoGPTPromptGeneratorTool,
+                    AutoGPTImageToMusicTool,
+                    AutoGPTDocumentAnsweringTool)
+from gradio_tools import GradioTool
+
+PromptGenerator = TypeVar("PromptGenerator")
+
+
+with open(str(Path(os.getcwd()) / ".env"), 'r') as fp:
+    load_dotenv(stream=fp)
+
+
+TOOLS = [
+    AutoGPTStableDiffusion(hf_token=os.getenv("GRADIO_TOOLS_HF_TOKEN")),
+    AutoGPTCaptioner(hf_token=os.getenv("GRADIO_TOOLS_HF_TOKEN")),
+    AutoGPTWhisperTool(hf_token=os.getenv("GRADIO_TOOLS_HF_TOKEN")),
+    AutoGPTTextToVideoTool(hf_token=os.getenv("GRADIO_TOOLS_HF_TOKEN")),
+    AutoGPTPromptGeneratorTool(hf_token=os.getenv("GRADIO_TOOLS_HF_TOKEN")),
+    AutoGPTDocumentAnsweringTool(hf_token=os.getenv("GRADIO_TOOLS_HF_TOKEN")),
+    AutoGPTImageToMusicTool(hf_token=os.getenv("GRADIO_TOOLS_HF_TOKEN")),
+    AutoGPTClipInterrogatorTool(hf_token=os.getenv("GRADIO_TOOLS_HF_TOKEN"))
+]
+
+def get_tool(tool: str) -> GradioTool:
+    return next(t for t in TOOLS if t.name == tool)
+
+
+class Message(TypedDict):
+    role: str
+    content: str
+
+
+class AutoGPTGradioTools(AutoGPTPluginTemplate):
+    """
+    Twitter API integrations using Tweepy
+    """
+
+    def __init__(self):
+        super().__init__()
+        self._name = "autogpt-gradio-tools"
+        self._version = "0.1.0"
+        self._description = "Calling Gradio Apps ."
+        ts = [t for t in os.getenv("AUTOGPT_GRADIO_TOOLS", "").split(",") if t != ""]
```

Comment:
Check if these exist before loading the plugin commands

Reply:
you can see how the email plugin does it for context

Reply:
Awesome - just pushed up this change!